### PR TITLE
Add pagination logic for setting the paging window

### PR DESCRIPTION
**Pull request by PatrickConnollyONS**
### What

Completing pagination logic

### How to review

Visual review of code and ensure all tests are passing

### Who can review

Anyone (someone familiar with frontend controllers would be well placed)

### CODE REVIEW

File: mapper/mapper_test.go [hunk 1]
```diff
@@ -229,7 +232,7 @@ func TestReleaseCalendarMapper(t *testing.T) {
 		}
 
 		Convey("CreateReleaseCalendar maps correctly to a model Calendar object", func() {
-			calendar := CreateReleaseCalendar(basePage, params, releaseResponse)
+			calendar := CreateReleaseCalendar(basePage, params, releaseResponse, config.Config{DefaultMaximumSearchResults: 1000})
```
Comment (rafahop) [inline mapper/mapper_test.go:235]: Maybe we could define this `config` in the outer convey (setup/given) for easier maintenance?

File: mapper/mapper_test.go [hunk 2]
```diff
@@ -289,7 +292,7 @@ func TestReleaseCalendarMapper(t *testing.T) {
 	})
 }
 
-// assertLinks checks that the actual model Link content is equal to the expected release Link
+// assertLinks checks that the actual model Link content is equal to the exStart release Link
```
Comment (rafahop) [inline mapper/mapper_test.go:295]: Not sure if this change is intentional? Shouldn't it remain `execpted`?

File: mapper/mapper_test.go [hunk 3]
```diff
@@ -299,7 +302,7 @@ func assertLinks(expected []releasecalendar.Link, actual []model.Link) {
 	}
 }
 
-// assertDateChanges checks that the actual model DateChanges content is equal to the expected release ReleaseDateChanges
+// assertDateChanges checks that the actual model DateChanges content is equal to the exStart release ReleaseDateChanges
```
Comment (rafahop) [inline mapper/mapper_test.go:305]: Same here

File: queryparams/params_test.go [hunk 4]
```diff
@@ -292,3 +292,21 @@ func TestDatesFromParams(t *testing.T) {
...
+		Convey("verify that the validated parameters are correctly returned in a url.Values mapping", func() {
+			uv := vp.AsQuery()
+			So(uv.Get(Limit), ShouldEqual, "10")
+			So(uv.Get(YearAfter), ShouldEqual, "2020")
+			So(uv.Get(MonthBefore), ShouldEqual, "12")
+			So(uv.Get(DayBefore), ShouldEqual, "31")
+			So(uv.Get(Keywords), ShouldEqual, "some keywords")
+			So(uv.Get(SortName), ShouldEqual, TitleAZ.String())
+			So(uv.Get(Published), ShouldEqual, "true")
+			So(uv.Get(Upcoming), ShouldEqual, "false")
```
[the remark this answers is not in the document]
Comment (PatrickConnollyONS) [reply queryparams/params_test.go:309]: Yes I was testing by type, i.e. if one date was set correctly it must be the others were too ,given they have the same defined type. But I have changed it to be clearer, and now test all fields as you suggest

**Review by rafahop — APPROVED** [review]

**Pull request merged**